### PR TITLE
[VDEVOPS-572] Custom ClusterRoles for roles in the organization

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: stable/sso-roles/templates/cluster-role-developers.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  # Read-only access to network policies
+  - apiGroups: ["networking.k8s.io"]
+    resources: ["networkpolicies"]
+    verbs: ["get", "list", "watch"]
+  # Access to view events to aid in debugging
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["get", "list", "watch"]
+  # Workload-specific access for custom resources, excluding potentially dangerous resources and restricting create and delete
+  - apiGroups: ["extensions", "argoproj.io", "aquasecurity.github.io", "external-secrets.io"]
```
Comment (ADobrodey) [inline stable/sso-roles/templates/cluster-role-developers.yaml]: Not sure about these extensions, in which scenarios they might be needed for developers?

File: stable/sso-roles/templates/cluster-role-developers.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    resources: ["*"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  # Restricted access to potentially dangerous resources such as secrets
+  - apiGroups: ["*"]
+    resources: ["secrets"]
+    verbs: ["get", "list", "watch"]
```
Comment (ADobrodey) [inline stable/sso-roles/templates/cluster-role-developers.yaml]: Not sure about "get" on secrets

File: stable/sso-roles/templates/cluster-role-maintainers.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    resources: ["horizontalpodautoscalers"]
+    verbs: ["*"]
+  # Full management of policy resources
+  - apiGroups: ["policy"]
+    resources: ["poddisruptionbudgets"]
+    verbs: ["*"]
+  # Node management
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["*"]
```
Comment (ADobrodey) [inline stable/sso-roles/templates/cluster-role-maintainers.yaml]: Maintainers are namespace restricted, not sure about full access to nodes

Comment (ghost) [reply stable/sso-roles/templates/cluster-role-maintainers.yaml]: In this case access to nodes api does not make sense. Removed

File: stable/sso-roles/templates/cluster-role-developers.yaml (hunk 1 above)
Comment (ghost) [reply stable/sso-roles/templates/cluster-role-developers.yaml]: removed

**Pull request closed without merging**